lib/buffer.js

```diff
@@ -65,6 +71,9 @@ function alignPool() {
  * would ever actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  if (!(this instanceof Buffer)) {
+    bufferDeprecationWarning();
+  }
```

Comment:
See https://github.com/nodejs/node/pull/8166 ... I'm working towards making this *more* consistent. If you look at the current implementation of the `internal/util` `printDeprecationMessage()` you'll see that it simply defers to the `process.emitWarning()` method already, so it's a bit of an unnecessary indirection.

Also, the way that `bufferDeprecationWarning()` is implemented in this PR is unnecessarily more complex than it needs to be. There's no reason to create a new object and closure.